In [1]:
import numpy as np
import pandas as pd
from tqdm import tqdm_notebook

In [2]:
PATH = "../data/"
data_full = pd.read_csv(f'{PATH}train.tsv', sep='\t')
data_full.head()

train_id                                 name  item_condition_id  \
0         0  MLB Cincinnati Reds T Shirt Size XL                  3   
1         1     Razer BlackWidow Chroma Keyboard                  3   
2         2                       AVA-VIV Blouse                  1   
3         3                Leather Horse Statues                  1   
4         4                 24K GOLD plated rose                  1   

                                       category_name brand_name  price  \
0                                  Men/Tops/T-shirts        NaN   10.0   
1  Electronics/Computers & Tablets/Components & P...      Razer   52.0   
2                        Women/Tops & Blouses/Blouse     Target   10.0   
3                 Home/Home Décor/Home Décor Accents        NaN   35.0   
4                            Women/Jewelry/Necklaces        NaN   44.0   

   shipping                                   item_description  
0         1                                 No description yet  
1         0  This keyboard is in great condition and works ...  
2         1  Adorable top with a hint of lace and a key hol...  
3         1  New with tags. Leather horses. Retail for [rm]...  
4         0          Complete with certificate of authenticity

In [3]:
len(data_full)

1482535

In [15]:
data_full.isnull().sum()

train_id                  0
name                      0
item_condition_id         0
category_name          6327
brand_name           632682
price                     0
shipping                  0
item_description          4
dtype: int64

# Categories

### Replace NaN in category_name by 'No Category'

In [5]:
print(data_full.category_name.isnull().sum(), 'items have no category')

6327 items have no category


In [6]:
category_name_replaced = data_full.category_name.fillna('No Category')

In [7]:
category_name_replaced.isnull().sum()

0

### Analysis

In [8]:
print("There is {} unique categories".format(category_name_replaced.nunique()))

There is 1288 unique categories


In [10]:
cat_counts = {} # dict to store number of items posted in each category
for cat in tqdm_notebook(category_name_replaced.unique()):
    cat_counts[cat] = len(category_name_replaced == cat)

In [11]:
cat_counts_df = pd.DataFrame(data={'category_name': list(cat_counts.keys()), 'count': list(cat_counts.values())})

In [12]:
cat_counts_df.head()

category_name    count
0                                  Men/Tops/T-shirts  1482535
1  Electronics/Computers & Tablets/Components & P...  1482535
2                        Women/Tops & Blouses/Blouse  1482535
3                 Home/Home Décor/Home Décor Accents  1482535
4                            Women/Jewelry/Necklaces  1482535

In [13]:
cat_counts_df.describe()

count
count     1288.0
mean   1482535.0
std          0.0
min    1482535.0
25%    1482535.0
50%    1482535.0
75%    1482535.0
max    1482535.0

In [14]:
print(len(cat_counts_df.loc[cat_counts_df['count'] == 1]), 'categories contain only a single item.')

0 categories contain only a single item.


### Conlusions

1288 unique categories, 6327 items have no category, 84 categories contain only a single item

# Item description

### Conlusions

In [18]:
(data_full.item_description == 'No description yet').sum()

82489

Although only 4 items have NaN in the description, 82,489 items have text saying 'No description yet'